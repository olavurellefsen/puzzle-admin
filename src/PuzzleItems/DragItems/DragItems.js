import React from "react";
import { DragItemsContainer, DragItemsImage } from "./DragItems.style";

export default () => (
  <DragItemsContainer data-testid="DragItemsContainer">
    <DragItemsImage src={"https://via.placeholder.com/400x800?text=DragItems"} data-testid="DragItemsImage" />
  </DragItemsContainer>
);