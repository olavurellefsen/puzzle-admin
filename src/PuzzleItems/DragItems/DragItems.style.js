import styled from 'styled-components';

export const DragItemsContainer = styled.div`
  min-width: 294px;
  overflow: hidden;
`;

export const DragItemsHeader = styled.div`
  text-align: left;
  padding-top: 10px;
  padding-left: 20px;
  font-size: 16px;
  color: #666666;
`;

export const DragItemsBox = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  padding-right: 14px;
`;

export const DragPlaceholder = styled.div`
  height: 200px;
  width: 200px;
  margin: 20px 20px;
`;

export const AddDragItem = styled.div`
  height: 200px;
  width: 200px;
  margin: 20px 20px;
  font-size: 32px;
  text-align: center;
  cursor: pointer;
`;