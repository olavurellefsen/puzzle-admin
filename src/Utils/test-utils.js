import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
