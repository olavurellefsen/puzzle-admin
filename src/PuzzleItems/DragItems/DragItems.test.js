import React from "react";
import { render, cleanup } from 'test-utils';
import DragItems from "./DragItems";
afterEach(cleanup)

it("should render correctly", () => {
  const { container } = render(<DragItems />);
  expect(container).toMatchSnapshot();
});

it("should show the DragItemsContainer element", () => {
  const { queryByTestId } = render(<DragItems />);
  expect(queryByTestId("DragItemsContainer")).toBeTruthy();
});

it("should show the DragItemsImage element", () => {
  const { queryByTestId } = render(<DragItems />);
  expect(queryByTestId("DragItemsImage")).toBeTruthy();
});