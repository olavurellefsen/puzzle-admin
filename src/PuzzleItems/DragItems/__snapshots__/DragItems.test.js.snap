// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa fePDys"
    data-testid="DragItemsContainer"
  >
    <img
      class="sc-bwzfXH bFHdpU"
      data-testid="DragItemsImage"
      src="https://via.placeholder.com/400x800?text=DragItems"
    />
  </div>
</div>
`;
